intrahospital_api/update_demographics.py

```diff
@@ -194,6 +195,103 @@ def update_if_changed(instance, update_dict):
         instance.save()
 
 
+def get_mrn_and_date_from_merge_comment(merge_comment):
+    """
+    Takes in an merge comment e.g.
+    " Merged with MRN 123456 Oct 18 2014 11:03AM  Merged with MRN 234567 on Oct 22 2013  4:44PM"
+    returns a list of (MRN, datetime_merged,)
+    """
+    regex = r'Merged with MRN (?P<mrn>\w*\d*) on (?P<month>\w\w\w)\s(?P<day>[\s|\d]\d) (?P<year>\d\d\d\d)\s(?P<HHMM>[\s|\d]\d:\d\d)(?P<AMPM>[A|P]M)'
+    found = list(set(re.findall(regex, merge_comment)))
+    result = []
+    for match in found:
+        mrn = match[0]
+        date_str = f"{match[2]} {match[1]} {match[3]} {match[4]}{match[5]}"
+        merge_dt = datetime.datetime.strptime(date_str, "%d %b %Y %I:%M%p")
+        result.append((mrn, timezone.make_aware(merge_dt),))
+    # return by merged date
+    return sorted(result, key=lambda x: x[1], reverse=True)
+
+
+def get_active_mrn_and_merged_mrn_data(mrn):
+    """
+    For an MRN return the active MRN related to it (which could be itself)
+    and a list of inactive MRNs that are associated with it.
+
+    If there are no inactive MRNs or we are to say which is the
+    active mrn return the MRN passed in and an empty list.
+
+    Returns all merged MRNs related to the MRN including the row
+    for the MRN from the CRS_Patient_Masterfile.
+
+    The merged comments can be nested for for MRN x
+    we can have MERGE_COMMENT "Merged with y on 21 Jan"
+    Then for y we can have the merge comment "Merged with z on 30 Mar"
+    This will return the rows for x, y and z
+
+    If there are no related rows, ie the patient
+    is not merged, return None.
+    """
+    query = """
+    SELECT *
+    FROM CRS_Patient_Masterfile
+    WHERE Patient_Number = @mrn
+    AND MERGED = 'Y'
+    AND MERGE_COMMENTS <> ''
+    AND MERGE_COMMENTS is not null
+    """
+    query_result = api.execute_hospital_query(
+        query, {"mrn": mrn}
+    )
+    if not query_result:
+        return mrn, []
+
+    # The merged comments can be nested for for MRN x
+    # we can have MERGE_COMMENT "Merged with y on 21 Jan"
+    # then for y we can have the merge comment "Merged with z on 30 Mar".
+    # So we traverse the merge comments and create a dictionary of MRN: row
+    # for every MRN related to the one passed in.
+    mrn_to_row = {mrn: query_result}
+    related_mrns = [i[0] for i in get_mrn_and_date_from_merge_comment(query_result["MERGE_COMMENTS"])]
+    for related_mrn in related_mrns:
+        related_result = api.execute_hospital_query(
+            query, {"mrn": related_mrn}
+        )
+        mrn_to_row[related_mrn] = related_result
+        related_related_mrns = [
+            i[0] for i in get_mrn_and_date_from_merge_comment(related_result["MERGE_COMMENTS"])
+        ]
+        for related_related_mrn in related_related_mrns:
```

Comment:
Use recursion ? 
(If nesting can happen one level, is there any reason it can't happen a second level?)